presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedJoinQueriesWithDynamicFiltering.java

```diff
@@ -0,0 +1,122 @@
+    }
+
+    @Test
+    public void testJoinWithEmptyBuildSide()
+    {
+        Session session = Session.builder(getSession())
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, FeaturesConfig.JoinDistributionType.BROADCAST.name())
+                .build();
+        DistributedQueryRunner runner = (DistributedQueryRunner) getQueryRunner();
+        ResultWithQueryId<MaterializedResult> result = runner.executeWithQueryId(
+                session,
+                "SELECT * FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey AND orders.totalprice = 123.4567");
+        assertEquals(result.getResult().getRowCount(), 0);
+
+        OperatorStats probeStats = searchScanFilterAndProjectOperatorStats(result.getQueryId(), "lineitem");
+        // Probe-side is not scanned at all, due to dynamic filtering:
+        assertEquals(probeStats.getInputPositions(), 0L);
+    }
+
+    @Test
+    public void testJoinWithSelectiveBuildSide()
+    {
+        Session session = Session.builder(getSession())
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, FeaturesConfig.JoinDistributionType.BROADCAST.name())
+                .build();
+        DistributedQueryRunner runner = (DistributedQueryRunner) getQueryRunner();
+        ResultWithQueryId<MaterializedResult> result = runner.executeWithQueryId(
+                session,
+                "SELECT * FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey AND orders.custkey = 1");
+        assertGreaterThan(result.getResult().getRowCount(), 0);
+
+        OperatorStats probeStats = searchScanFilterAndProjectOperatorStats(result.getQueryId(), "lineitem");
+        // Probe side may be partially scanned, depending on the drivers' scheduling:
+        assertLessThanOrEqual(probeStats.getInputPositions(), countRows("lineitem"));
+    }
+
+    private OperatorStats searchScanFilterAndProjectOperatorStats(QueryId queryId, String tableName)
+    {
+        DistributedQueryRunner runner = (DistributedQueryRunner) getQueryRunner();
+        Plan plan = runner.getQueryPlan(queryId);
+        PlanNodeId nodeId = PlanNodeSearcher.searchFrom(plan.getRoot())
+                .where(node -> {
+                    if (!(node instanceof ProjectNode)) {
+                        return false;
+                    }
+                    ProjectNode projectNode = (ProjectNode) node;
+                    FilterNode filterNode = (FilterNode) projectNode.getSource();
+                    TableScanNode tableScanNode = (TableScanNode) filterNode.getSource();
+                    return tableName.equals(((HiveTableHandle) (tableScanNode.getTable().getConnectorHandle())).getTableName());
+                })
+                .findOnlyElement()
+                .getId();
+        return runner.getCoordinator()
+                .getQueryManager()
+                .getFullQueryInfo(queryId)
+                .getQueryStats()
+                .getOperatorSummaries()
+                .stream()
```

Comment:
Move all these to the same line with the previous line

Reply:
It will exceed our limit in one line

Reply:
We don't have line break limit. Let's follow the coding style to break at `.stream()` only.